src/components/plots/areaPlot.ts

```diff
@@ -146,10 +147,14 @@ export module Plots {
         return;
       }
 
+      if (this._constantBaselineValueProvider != null) {
+        yScale.removePaddingExceptionsProvider(this._constantBaselineValueProvider);
+        this._constantBaselineValueProvider = null;
```

Comment:
This logic doesn't seem correct. The previous logic removed the provider when `constantBaseline` was null, rather than when the provider was null.

With the new logic, if I set a constant baseline, then change `y0()` so the baseline isn't constant, the padding exception isn't removed.